src/Xamarin.Android.Build.Tasks/Tasks/GetAssetPacks.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.Android.Build.Tasks;
+
+namespace Xamarin.Android.Tasks
+{
+	// We have a list of files, we want to get the
+	// ones that actually exist on disk.
+	public class GetAssetPacks : AndroidTask
+	{
+		public override string TaskPrefix => "GAP";
+
+		[Required]
+		public ITaskItem[] Assets { get; set; }
+
+		[Required]
+		public ITaskItem IntermediateDir { get; set; }
+
+		public string[] MetadataToCopy { get; set; } = { "DeliveryType" };
+
+		[Output]
+		public ITaskItem[] AssetPacks { get; set; }
+
+		public override bool RunTask ()
+		{
+			Dictionary<string, ITaskItem> assetPacks = new Dictionary<string, ITaskItem> ();
+			Dictionary<string, List<string>> files = new Dictionary<string, List<string>> ();
+			foreach (var asset in Assets)
+			{
+				var assetPack = asset.GetMetadata ("AssetPack");
+				if (string.IsNullOrEmpty (assetPack) || string.Compare (assetPack, "base", StringComparison.OrdinalIgnoreCase) == 0)
+					continue;
+				if (!assetPacks.ContainsKey (assetPack))
+				{
+					assetPacks[assetPack] = new TaskItem (assetPack);
+					assetPacks[assetPack].SetMetadata ("AssetPack", assetPack);
+					assetPacks[assetPack].SetMetadata ("AssetPackCacheFile", Path.Combine (IntermediateDir.ItemSpec, assetPack, "assetpack.cache"));
+				}
+				var item = assetPacks[assetPack];
+				foreach (var metadata in MetadataToCopy)
+					if (string.IsNullOrEmpty (item.GetMetadata (metadata)))
+						item.SetMetadata (metadata, asset.GetMetadata (metadata));
+				if (!files.ContainsKey (assetPack))
+					files[assetPack] = new List<string> ();
+				files[assetPack].Add (asset.ItemSpec);
+			}
+
+			foreach (var kvp in assetPacks) {
+				// write out the file cache list
+				// write out the metadata as well.
+				ITaskItem item = kvp.Value;
+				var cacheFile = kvp.Value.GetMetadata ("AssetPackCacheFile");
+				var sb = new StringBuilder ();
+				foreach (var file in files [kvp.Key])
+					sb.AppendLine ($"{file}:{File.GetLastWriteTimeUtc (file)}");
+				sb.AppendLine (item.GetMetadata ("DeliveryType") ?? "InstallTime");
+				Files.CopyIfStringChanged (sb.ToString (), cacheFile);
```

Comment:
Could this use the other pattern instead of `StringBuilder`? It should allocate a lot less:

https://github.com/xamarin/xamarin-android/blob/384e1a76abf4798ceec1648d124860276b222d33/src/Xamarin.Android.Build.Tasks/Utilities/RtxtWriter.cs#L16-L22